dhcpv4/nclient4/example_lease_test.go

```diff
@@ -0,0 +1,37 @@
+package nclient4_test
+
+import (
+	"context"
+	"log"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv4/nclient4"
+)
+
+func Example() {
+	ifname := "enp0s10"
+	clntid := "client-1"
+	var idoptlist dhcpv4.OptionCodeList
+	//specify option61 is part of client identification used by DHCPv4 server
+	idoptlist.Add(dhcpv4.OptionClientIdentifier)
+	clntOptions := []nclient4.ClientOpt{nclient4.WithClientIDOptions(idoptlist), nclient4.WithDebugLogger()}
```

Comment:
I do not really like this API and how some options are copied around.

This is how it looks like now:

```
// it tells the client to copy some options to the lease struct
clntOptions := []nclient4.ClientOpt{nclient4.WithClientIDOptions(idoptlist)}

// create the option you are interested in
option61 := dhcpv4.OptClientIdentifier([]byte(clntid))

// tell the Request to add the option you are interested in
lease, err := clnt.Request(context.Background(), dhcpv4.WithOption(option61))
```

Can it be changed to:

```
// create the option you are interested in
option61 := dhcpv4.OptClientIdentifier([]byte(clntid))

// tell the Request to add the option you are interested in
lease, err := clnt.Request(context.Background(), dhcpv4.WithOption(option61))

// tell the Release to add the option you are interested in
clnt.Release(lease, dhcpv4.WithOption(option61))
```

I think it will be more consistent.

I am not going to block it though if @hugelgupf was ok with the previous design

Reply:
I had a similar concern, but tbh not the time to follow up. Either is fine with me, this could be nicer.

I haven't fully followed where this ID option first comes into play. Perhaps that could also be stored in the Lease?

Reply:
(I'm guessing the expectation is that it be the same as in the original Request or something?)

Reply:
The reason behind Lease.IDOptions is in some cases DHCP server uses certain DHCP options to identify a lease, for example, RFC2131 allows DHCP server use "Client Identifier" option61 to identify a lease (just like in example_lease_test.go), so when client need to release lease, it needs to use same identifier so that server could locate the lease, but option61 is not included in ACK/Offer, which means it needs to be part of lease, that's why Lease has this IDOptions;  I know there is DHCP server in some other use cases use different option like vendor class + client id;
hopefully this clarify the use case

Reply:
As a user of the client library you need to create any additional option and pass it with a modifier to `Request`. What is them problem of passing it the same way to `Release`?

Reply:
but in following case, the client program might not aware the identification options if they are not part of lease:
1. client program run 1st time, doing DORA for 1000 with 1000 different option61, get 1000 leases, save them to a file, and then quit
2. later on, client program need to run again to release these 1000 leases, so it will read leases from file, but at this point client  program doesn't know what or even if option61 need to be included in Release unless option61 is part of lease

Reply:
I think an abstraction on top of this client should handle the persistence of the lease in the way you want.

You will be creating the additional options and you can persist them.

This is not about option61, which is probably not what you are using because the library does not implement it.

Or even have a configuration file with all the additional options that are needed. I would leave all of this to an abstraction on top of this client.

Other than how these options copied around, the diff looks good to me.